tests/integration/test_upgrades.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Superset charm upgrades integration tests."""
+
+import asyncio
+import logging
+
+import pytest
+import pytest_asyncio
+import requests
+from helpers import (
+    APP_NAME,
+    METADATA,
+    POSTGRES_NAME,
+    REDIS_NAME,
+    SUPERSET_SECRET_KEY,
+    get_unit_url,
+    perform_superset_integrations,
+)
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.skip_if_deployed
+@pytest_asyncio.fixture(name="deploy", scope="module")
+async def deploy(ops_test: OpsTest):
+    """Deploy the app."""
+    asyncio.gather(
+        ops_test.model.deploy(POSTGRES_NAME, channel="14", trust=True),
+        ops_test.model.deploy(REDIS_NAME, channel="edge", trust=True),
```

Comment:
This is a more general question. Does it make sense to do your integration tests against the `edge` channel of a component?

Reply:
Unfortunately the redis `stable` channel is not compatible with the `data platform libs` library that is used across the charms we create. So the `edge` channel is actually the one we have to use for deployments as is. Once it's promoted to stable then we can go back and change this.